Objects/Converters/ConverterRevit/ConverterRevitShared/Partial Classes/ConvertFloor.cs

```diff
@@ -196,38 +196,68 @@ private List<ICurve> GetProfiles(DB.CeilingAndFloor floor)
     }
 
     // in order to create a revit floor, we need to pass it a flat profile so change all the z values
-    private void ChangeZValuesForCurves(ICurve curve, double z)
+    private ICurve GetFlattenedCurve(ICurve curve, double z)
     {
+      // kind of a hack. Editing our speckle objects is so much easier than editing the revit objects
+      // so scale this z value from the model units to the incoming commit units and then those values will
+      // get scaled back to the model units when this entire outline gets scaled to native
+
       switch (curve)
       {
         case OG.Line line:
-          // kind of a hack. Editing our speckle objects is so much easier than editing the revit objects
-          // so scale this z value from the model units to the incoming units and then they will get scaled
-          // back to the model units when this entire outline gets scaled to native
-          line.start.z = z * Speckle.Core.Kits.Units.GetConversionFactor(ModelUnits, line.start.units);
-          line.end.z = z * Speckle.Core.Kits.Units.GetConversionFactor(ModelUnits, line.end.units);
-          return;
+          return new OG.Line(
+            new OG.Point(
+              line.start.x,
+              line.start.y,
+              z * Speckle.Core.Kits.Units.GetConversionFactor(ModelUnits, line.start.units),
+              line.start.units),
+            new OG.Point(
+              line.end.x,
+              line.end.y,
+              z * Speckle.Core.Kits.Units.GetConversionFactor(ModelUnits, line.end.units),
+              line.end.units),
+            line.units
+            );
```

Comment:
I suggest getting the conversion factor just once for the line (point units should always be the same as the line units anyway)